src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt

```diff
@@ -523,22 +523,6 @@ class KotlinCompilation {
 
 	/** Performs the 3rd compilation step to compile Kotlin source files */
 	private fun compileKotlin(sourceFiles: List<File>): ExitCode {
-
-		/**
-		 * Here the list of compiler plugins is set
-		 *
-		 * To avoid that the annotation processors are executed twice,
-		 * the list is set to empty
-		 */
-		MainComponentRegistrar.threadLocalParameters.set(
-			MainComponentRegistrar.Parameters(
-				listOf(),
-				KaptOptions.Builder(),
-				compilerPlugins
-			)
-		)
-
-
```

Comment:
Why did you remove this again? I think we always need to add the compiler plugins to the `compileKotlin` step, even if they already ran in the `stubsAndApt` step.

Reply:
H @tschuchortdev , i removed it because i wanted to change a less as possible, because i couldn't guarantee that it would break any behaviour of any other compiler plugins that runs in any of the compiler phases. All i can say is, that i could test my MpApt Plugin with these changes and it didn't break any of the existing tests and Kapt was also working. I don't know much about when which compiler phase is running, because my plugin is only running in one phase.

Reply:
Are you creating new source files with your plugin or simply adding to the AST?

Reply:
I'm creating new source files

Reply:
My plugins just detects the annotations, it doesn't provide code generation

Reply:
I think we still need to run the plugins in the `compileKotlin` step then, because all the changes done to the AST during `stubsAndApt` will be discarded, so a compiler plugin that changes the AST would have no effect.  

But I think we also need to run the plugins in the `stubsAndApt` step as well, before kapt plugin is run.

Reply:
I also think that we should add it to both. I could restore my deleted version.

Reply:
Sounds good 👍

Reply:
I restored it, and changed the check in stubsAndApt to "annotationProcessors.isEmpty()" again, because without, all added plugins were executed in both phases. In the future we need an option where you can set in which phases you want to execute the plugins, but for now i would keep it that way. If you think otherwise, just add " && compilerPlugins.isEmpty()" to the if statement again.